code/game/turfs/change_turf.dm

```diff
@@ -25,9 +25,12 @@ GLOBAL_LIST_INIT(blacklisted_automated_baseturfs, typecacheof(list(
 		copy_to_turf.icon_state = icon_state
 	if(copy_to_turf.icon != icon)
 		copy_to_turf.icon = icon
-	if(color)
+	if(atom_colours?.len)
```

Comment:
is this a consequence of the color filter stuff? also you should use LAZYLEN here

Reply:
uh I don't understand your question.

Point is if somehow, someone passes here turf, colored via color var, then it runtimes and brakes.
It's okay with TG codebase, because you keep it kinda clean, but for example our geniuses painted asteroids turfs, and as result evac shuttle asteroid with strapped engines throws runtime here and as result shuttle travels without any walls at all.

So ugh, I moved that asteroid paint to add_atom_colour proc, but here we are.